css-view-transitions-2/Overview.bs

```diff
@@ -0,0 +1,373 @@
+
+<xmp class=idl>
+		[Exposed=Window]
+		interface PageRevealEvent : Event {
+			readonly attribute ViewTransition? viewTransition;
+		};
+</xmp>
+
+The <dfn attribute for=PageRevealEvent>viewTransition</dfn> [=getter steps=] are to return the
+[=inbound cross-document view-transition=] for [=this's=] [=relevant global object's=] [=associated document=].
+
+
+
+# Algorithms # {#algorithms}
+
+## Additions to {{ViewTransition}} ## {#view-transitions-extension}
+
+	A {{ViewTransition}} additionally has:
+	<dl dfn-for=ViewTransition>
+		: <dfn>is cross-document</dfn>
+		:: a boolean, initially false.
+	</dl>
+
+## Monkey patches to HTML ## {#monkey-patch-to-html}
+
+	<div algorithm="monkey patch to unload">
+		Run the following step before <a href="https://html.spec.whatwg.org/multipage/document-lifecycle.html#unloading-documents:update-the-visibility-state">updating the visibility state</a> in the [=unload=] steps:
+
+		1. If |newDocument| is given, then [=setup outbound cross-document view transition=] given |oldDocument| and |newDocument|.
+	</div>
+
+	<div algorithm="monkey patch to render blocking">
+		Run the following step at the end of [=unblock rendering=]:
+
+		1. [=reveal document|reveal=] |document|.
+
+		Reword the text in the [=render-blocked=] definition, to call [=reveal document=] |document| if the [=implementation-defined=] timeout value has been exceeded.
+	</div>
+
+	<div algorithm="monkey patch to reactivation">
+		Run the following step in [=Document/reactivate=], before step 4 (querying for [=page showing=]):
+
+		1. [=reveal document|reveal=] |document|.
+	</div>
+
+	<div algorithm="page reveal">
+	To <dfn>reveal {{Document}}</dfn> |document|:
+		1. Assert: |document|'s [=page showing=] is false.
+
+		1. If |document| is [=render-blocked=] then return.
+
+		1. Fire a new event named <code>reveal</code> on |document|'s [=relevant global object=],
+			using {{PageRevealEvent}}.
+
+		1. Let |transition| be the result of getting the [=inbound cross-document view-transition=] for |document|.
+
+		1. If |transition| is not null, then [=activate view transition|activate=] |transition|.
+	</div>
+
+## Setting up and activating the cross-document view transition
+
+	<div algorithm>
+		To <dfn>setup outbound cross-document view transition</dfn> given a {{Document}} |oldDocument|
+		and a {{Document}} |newDocument|:
+
+		1. Assert: this is called when |oldDocument| is [=unloading=].
+
+		1. If |oldDocument|'s [=active view transition=] is not null,
+			then [=skip the view transition|skip=] |oldDocument|'s [=active view transition=]
+			with an "{{AbortError}}" {{DOMException}} in |oldDocument|'s [=relevant Realm=].
+
+		1. If |oldDocument|'s [=environment settings object/origin=] is not [=same origin=] as
+			|newDocument|'s [=environment settings object/origin=] then return.
+
+		1. If |newDocument| [=was created via cross-origin redirects=] is true
```

Comment:
OK this is important and I gave this a lot of thought.

For an A->B->C case, the previous condition would be true anyway, so we would return early. What matters here is an A->B->A redirect ("confused agent"). For that, see the next line and note. In a BFCache restore/prerender reactivation, `|newDocument|'s [=latest entry=]` is not null so we would ignore the cross-origin redirect. 

The restriction on cross-origin redirects only applies *when the redirect is happening*, meaning in the initial navigation. This is to prevent a situation where the view transition obscures the fact that there was a redirect.

From that point on, in BFCache restores and in prerender activations, the cross-origin redirects are no longer a factor. This is simply a navigation between same-origin documents.